gcloud-java-pubsub/src/main/java/com/google/gcloud/pubsub/PubSub.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.pubsub;
+
+import com.google.gcloud.Page;
+import com.google.gcloud.Service;
+
+import java.util.List;
+import java.util.concurrent.Future;
+
+/**
+ * An interface for Google Cloud Pub/Sub.
+ *
+ * @see <a href="https://cloud.google.com/pubsub/">Google Cloud Pub/Sub</a>
+ */
+public interface PubSub extends Service<PubSubOptions> {
+
+  class ListOption {
+    // page size
+    // page token
+  }
+
+  class PullOption {
+    // bool return_immediately = 2;
+    // int32 max_messages = 3;
+  }
+
+  // topics
+  //////////////////////
+  Topic createTopic(TopicInfo topic);
+
+  Future<Topic> createTopicAsync(String topic);
+
+  // null if not found
+  Topic getTopic(String topic);
+
+  Future<Topic> getTopicAsync(String topic);
+
+  // false if not found
+  boolean deleteTopic(String topic);
+
+  Future<Boolean> deleteTopicAsync(String topic);
+
+  Page<Topic> listTopics(ListOption... options);
+
+  // todo: consider AsyncPage that has nextPageAsync
+  Future<Page<Topic>> listTopicsAsync(ListOption... options);
```

Comment:
Yes my point is that if we use `Page` as it is now only the first call to list can be asynchronous, fetching pages after the first one would be synchronous. This is solved by having `AsyncPage` with its own `Future<Page<T>> nextPageAsync()` as you suggested.

`valuesAsync()` has a slightly strange semantics. Only the first page is fetched asynchronously while iterating through the values might cause synchronous calls. Not sure this is a problem or not.